### PR TITLE
Enforce compute cluster version

### PR DESCRIPTION
**Pull request by KuuCi**
To resolve:
https://databricks.slack.com/archives/C06RXC1NPQC/p1727371779924089

Essentially, we check if the compute runtime version is higher than our `databricks-connect` version to determine whether to use 'dbconnect' or not, but we don't check if the compute runtime version is _too high_. There is no formula to determine which version is too high, so we should just enforce the customer using a computer that has the same major runtime as the `databricks-connect` version

Test:
Error should be thrown:
image
with the following compute:
image

But should work:

### CODE REVIEW

File: llmfoundry/command_utils/data_prep/convert_dataset_hf.py [hunk 1]
```diff
@@ -451,6 +451,7 @@ def convert_dataset_hf_from_args(
         ValueError: If the output directory already contains the requested splits
         ValueError: If `concat_tokens` is set but `tokenizer` is not
     """
+    os.environ['WORLD_SIZE'] = '1'
```
Comment (dakinggg) [inline llmfoundry/command_utils/data_prep/convert_dataset_hf.py]: are these changes supposed to be included?

**Review by dakinggg — COMMENTED** [review]
Is this check potentially too restrictive? do we _actually_ need an exact match? or we just don't know?

File: llmfoundry/command_utils/data_prep/convert_dataset_hf.py (hunk 1 above)
Comment (KuuCi) [reply llmfoundry/command_utils/data_prep/convert_dataset_hf.py]: Actually going to close this pr, during testing seems like every version I tested worked (strangely), so perhaps old bug was actually transient, and was solved not by the version change but simply restarting the cluster. Will revisit if it comes up again.

**Comment by KuuCi** [comment]
Not needed as all versions currently work... see above comment.

File: llmfoundry/command_utils/data_prep/convert_dataset_hf.py (hunk 1 above)
Comment (KuuCi) [reply llmfoundry/command_utils/data_prep/convert_dataset_hf.py]: Original motivation is we don't know what works and what gets deprecated

**Pull request closed without merging**